### ISSUE TITLE
(Suggestion) An alternative to faction auto-balancing 

### ISSUE THREAD

**Issue by BanditoDoritoIsNeato** [HaloSpaceStation/HaloSpaceStation13#1566]
People were talking about faction auto-balancing in the Discord and I had an alternative to put forward. The overall suggestion I had was posted there as part of the discussion, and a copy of it is here below for review.

"_Look into putting a serverside system that counts the number of players in roles. Weigh the roles based on their effectiveness within a numerical system (ex: ODSTs weigh 3, muhreens 2, crewman 1, jackals 2 or 3) and compare the factions to see if one or several are severely outmatched, such as in cases where there are 15 UNSC vs 2 covenant. After the round has been going for X amount of time to allow for new joiners, the system gives the outmatched faction the opportunity to play as a special role, like those two covenant players get to play as a bonded pair of hunters or something. Admins/mods or whatever have commands they can put in to circumvent, disable, or manually influence the system like the bots on this discord have, based on what level of accessibility is suitable. Allows for the server to make basic judgement calls in lowpop when admins aren’t around, and gives the hopeless faction something cool to work with that can even the odds or just make it a fun loss._"

Basically, I didn't see forcing players to play certain factions/roles due to balancing teams as being a safe or benign thing given not everyone knows the role or wants to play it. It could easily result in a lot of greentide antics that people won't be very welcoming to, potentially including the salty kind since people can't play what they want. With that in mind, this is my initial suggestion on fixing the core problem of team balancing a different way.

Here's a more refined description from what was said on Discord

>>>>>>>Primary Functionality
--Balance the teams by temporary use of special roles, making the outnumbered players more powerful rather than evening the population, and giving them something interesting even if they still lose.

1. Create a serverside system that numerically weighs each faction's occupied roles in relation to how powerful the roles are. Count stuff with values attached, remember the numbers.
2. Compare the overall weight of one faction to another, deciding if one or more is severely outmatched - like that of fifteen UNSC being opposed by two covenant. 
3. Wait until the round has been active for X amount of time to allow new players to join, potentially balancing it out by itself. 
4. If that does not occur after the time has been exceeded, the server notifies the underdogs that there are special roles available to them due to the conditions being met. It pings them like an admin message or server vote to reply yes or no to an opt-in statement.
5. On no, the system does nothing with that player. On yes, the system respawns them instantly as the special role.

---The underdog players can decide to take the role if they so choose, or just decline in order to do their own thing. It doesn't force anything because there will always be players that *do* want to try lone-wolfing. 
---Purely for example, those two covenant players could become a bonded pair of hunters because of the system's intervention, pitting them against the 15 UNSC players.

>>>>>>>Secondary Functionality
---It would definitely need manual commands for the admins/mods to circumvent, disable, or manually influence it. I assume you'll want different ranks of staff to have different access with these commands too. 
---Having the capability to assign and easily edit loadouts or presets for the system to use when spawning things, similar to how roles give you gear on spawn. You'll want the efficiency and control.
---A function that lets the admins/mods designate spawn points on the fly for manual control as well as permanently to assign default locations. Gives you more control. 

There's also the potential for minor intelligence to be added with that last one in the future. It could be upgraded to use sets of spawns to place the special roles in the path of the other factions. That would be a topic for a long time from now though.

>>>>>>>Other
---Could be a good way of testing special roles. Gives people stuff to play with in the process, makes it interesting for both sides too.

It's literally midnight at the time of writing this, so I'll post and sleep on the idea to see if I can come up with more to add to it. I'll see about addressing any problems you guys find with the idea too sometime tomorrow.

(Side note; first time posting on GitHub. Don't know the interface or what's expected, so if I'm doing something wrong with the post I'll fix that tomorrow too.)

**Comment by BanditoDoritoIsNeato** [comment]
Yep, text in my post is weird. I'll see about fixing it tomorrow.

**Comment by BanditoDoritoIsNeato** [comment]
Text should be fixed now.

**Comment by CdrRiley** [comment]
vote on discord has shown the player base DOES NOT want an autobalance function

**Comment by BanditoDoritoIsNeato** [comment]
> vote on discord has shown the player base DOES NOT want an autobalance function

The Discord vote for a team-balance function was based on locking roles and forcing players to pick (or be directly placed in) the factions/roles with less players. The vote was overwhelmingly against that, yes, but that also means that the problem it attempted to fix still exists as nothing was changed as dictated by the vote outcome.

My suggestion here is to specifically NOT do that method the vote offered, as I believe that it would just cause more problems as I mentioned in my original post - I also voted "No" in the Discord vote. Instead, think about this one as a compromise.

**Comment by CdrRiley** [comment]
----------Secondary Functionality
---It would definitely need manual commands for the admins/mods to circumvent, disable, or manually influence it. I assume you'll want different ranks of staff to have different access with these commands too.
---Having the capability to assign and easily edit loadouts or presets for the system to use when spawning things, similar to how roles give you gear on spawn. You'll want the efficiency and control.
---A function that lets the admins/mods designate spawn points on the fly for manual control as well as permanently to assign default locations. Gives you more control.

There's also the potential for minor intelligence to be added with that last one in the future. It could be upgraded to use sets of spawns to place the special roles in the path of the other factions. That would be a topic for a long time from now though.

----------Other
---Could be a good way of testing special roles. Gives people stuff to play with in the process, makes it interesting for both sides too.

It's literally midnight at the time of writing this, so I'll post and sleep on the idea to see if I can come up with more to add to it. I'll see about addressing any problems you guys find with the idea too sometime tomorrow.

- coder wise no?
-spwan items would need to be chek and codes as well  for waht eaxtacl loadout?
-dont need to add a coustom spwan area you either spwan on your fraction base or admin area then get Tp for speical roles
-speical roles like ONI ERT have own spwan and ship

**Comment by BanditoDoritoIsNeato** [comment]
>coder wise no?
-spwan items would need to be chek and codes as well for waht eaxtacl loadout?
-dont need to add a coustom spwan area you either spwan on your fraction base or admin area then get Tp for speical roles
-speical roles like ONI ERT have own spwan and ship

---Spawn items like occupation loadouts do upon spawn/respawn. Do the same thing as that, just as a hidden set of loadouts only accessible through the system and it's settings. 
---The spawn location part is taking into consideration that the underdog players may be in the middle of something far from their usual spawn - like boarding a civilian ship as they cannot fight the other factions head-on - or separated along the way. It's intent is to put them where they need to be and to do so without the need for an admin to help it, as many of the original problems revolve around lowpop. Having special spawns for this in addition to the default ones would allow the system a degree of flexibility, potentially with something added later that can semi-intelligently place the special roles where they will cross paths with the opposing factions so that they can get straight into what they need to be doing. They might not even have a working ship too.
---That would be useful to incorporate into this for the same reason.
